### PR TITLE
Add nifty to conda-forge

### PR DESCRIPTION
**Pull request by constantinpape**
Add [nifty](https://github.com/DerThorsten/nifty), library for graph based image segmentation.
Pinging @DerThorsten.

This is a c++ library with python bindings, pinging.
@conda-forge/help-python-c
@conda-forge/help-c-cpp

### CODE REVIEW

**Comment by constantinpape** [comment]
Builds fail due to some build dependency clashes, gonna have a look at this tomorrow.

**Comment by constantinpape** [comment]
The azure builds seem to work fine, they just fail because the test discovery fails.
I will check that later.
@conda-forge/help-c-cpp
It looks like builds between azure and the other CI tools are redundant?
Is this a bug or should the build really trigger all CIs?

File: recipes/nifty/build.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        -DCMAKE_BUILD_TYPE=RELEASE \
+        -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}\
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DCMAKE_PREFIX_PATH=${PREFIX} \
+\
+        -DCMAKE_SHARED_LINKER_FLAGS="${LDFLAGS}" \
+        -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}" \
+        -DCMAKE_CXX_FLAGS="${CXXFLAGS}" \
+        -DCMAKE_CXX_FLAGS_RELEASE="${CXXFLAGS} -O3 -DNDEBUG" \
+        -DCMAKE_CXX_FLAGS_DEBUG="${CXXFLAGS}" \
```
Comment (xhochy) [inline recipes/nifty/build.sh]: I would have expected that you don't need any of the following lines:

```
        -DCMAKE_C_COMPILER=${CC} \
        -DCMAKE_CXX_COMPILER=${CXX} \
        -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}\
        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
        -DCMAKE_PREFIX_PATH=${PREFIX} \
\
        -DCMAKE_SHARED_LINKER_FLAGS="${LDFLAGS}" \
        -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}" \
        -DCMAKE_CXX_FLAGS="${CXXFLAGS}" \
        -DCMAKE_CXX_FLAGS_RELEASE="${CXXFLAGS} -O3 -DNDEBUG" \
        -DCMAKE_CXX_FLAGS_DEBUG="${CXXFLAGS}" \
```

Comment (constantinpape) [reply recipes/nifty/build.sh]: That's all left-overs from the recipe I forked that from, so I can try without.

**Comment by constantinpape** [comment]
Looks like linux builds are ok now and there are some build issues on mac and windows.
I will have a look at that later.

**Comment by DerThorsten** [comment]
@constantinpape  we could also remove the glpk dependencies from nifty, since glpk based multicut does not scale beyond miniscule examples

**Comment by constantinpape** [comment]
@DerThorsten good point, I removed it

File: recipes/nifty/build.sh (hunk 1 above)
Comment (constantinpape) [reply recipes/nifty/build.sh]: So it turns out we need `CMAKE_PREFIX_PATH` and `CMAKE_INSTALL_PREFIX`.
I removed the rest.

**Comment by constantinpape** [comment]
Linux builds are passing, but we still have some issues for mac and win. I will look into this in the next couple of days.
I am still a bit confused about all the CIs. Are these redundant or are they actually building the package for different python versions?

**Comment by mariusvniekerk** [comment]
So the various ci builds run for different platforms.  We are in the process of moving over fully to azure where we have substantially more build capacity.

**Comment by constantinpape** [comment]
> various ci builds run for different platforms

Yes, that's why I was a bit confused, because azure runs all three OS, while the others only run a single OS each and I couldn't really figure out if there was a difference between say azure linux and circleci linux.

> moving over fully to azure where we have substantially more build capacity

I see, is there a timeline on this?

**Comment by constantinpape** [comment]
Ok current issues:
- OSX (on travis and azure) fails due to type conversion issues in [this constructor](https://github.com/DerThorsten/nifty/blob/master/include/nifty/array/arithmetic_array.hxx#L124). We will fix this in the upstream project.
- Windows (appveyor): fails due to missing `std`s before `size_t`. We will fix this in the upstream project.
- Windows (azure): this build should be skipped, see [here](https://github.com/conda-forge/staged-recipes/blob/8b8d081f2d2f2e26c6b10ec68db2045dd4fb4669/recipes/nifty/meta.yaml#L14). I don't get why the build is failing.

**Comment by chrisburr** [comment]
> Windows (azure): this build should be skipped, see [here](https://github.com/conda-forge/staged-recipes/blob/8b8d081f2d2f2e26c6b10ec68db2045dd4fb4669/recipes/nifty/meta.yaml#L14). I don't get why the build is failing.

You can ignore the status of this CI job, the Azure pipelines are still experimental the Windows one in particular is known to be broken.

**Comment by constantinpape** [comment]
Linux and OSX builds pass now.
The appveyor windows build fails for the second (python3.7) build, because it cannot find the cmake file:
https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22467463#L9114

I have no idea why this happens. The first build (python 3.6) passes.

**Comment by constantinpape** [comment]
@conda-forge-admin please rerender

**Comment by xhochy** [comment]
@constantinpape `rerender` is not working on `staged-recipes`. You need to rebase on master to get the latest CI setup.

**Comment by xhochy** [comment]
@constantinpape You seemed to have done a merge and a rebase. This screwed the diff somehow :(

**Comment by constantinpape** [comment]
Somehow the rebase kicked out all my commits and then I pulled from my fork; which was probably wrong ...
I could just close this PR and reopen a clean version?

**Comment by xhochy** [comment]
@constantinpape no need to close this PR, you could locally create a new branch and just force-push to the equally named remote branch.

**Comment by constantinpape** [comment]
@xhochy  Good point.
I did that, now the diff looks reasonable again.

**Comment by constantinpape** [comment]
Linux and Mac builds look fine.
I don't understand what's going wrong with the Win builds.
Azure fails with 
```
CMake Error at CMakeLists.txt:4 (project):
  Failed to run MSBuild command:

    MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 4.7.3062.0
    [Microsoft .NET Framework, version 4.0.30319.42000]
    Copyright (C) Microsoft Corporation. All rights reserved.
    
    Build started 4/26/2019 8:43:02 AM.
    Project "C:\Miniconda\conda-bld\nifty_1556267858544\work\build\CMakeFiles\3.14.3\VCTargetsPath.vcxproj" on node 1 (default targets).
    C:\Miniconda\conda-bld\nifty_1556267858544\work\build\CMakeFiles\3.14.3\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "C:\Miniconda\conda-bld\nifty_1556267858544\work\build\CMakeFiles\3.14.3\VCTargetsPath.vcxproj" (default targets) -- FAILED.
    
    Build FAILED.
    
    "C:\Miniconda\conda-bld\nifty_1556267858544\work\build\CMakeFiles\3.14.3\VCTargetsPath.vcxproj" (default target) (1) ->
      C:\Miniconda\conda-bld\nifty_1556267858544\work\build\CMakeFiles\3.14.3\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    
        0 Warning(s)
        1 Error(s)
    
    Time Elapsed 00:00:00.85
    

  Exit code: 1
```

Appveyor falils in the second build with
```
A subdirectory or file build already exists.
CMake Error: The source directory "C:/bld/nifty_1556267563172/work" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
Error: could not load cache
Traceback (most recent call last):
  File ".ci_support\build_all.py", line 131, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 121, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\build.py", line 2314, in build_tree
    notest=notest,
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\build.py", line 1444, in build
    windows.build(m, build_file, stats=build_stats, provision_only=provision_only)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\windows.py", line 342, in build
    check_call_env(cmd, cwd=m.config.work_dir, stats=stats, rewrite_stdout_env=rewrite_env)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\utils.py", line 374, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "C:\Miniconda36-x64\lib\site-packages\conda_build\utils.py", line 354, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['cmd.exe', '/c', 'conda_build.bat']' returned non-zero exit status 1.
```

**Comment by constantinpape** [comment]
@conda-forge-admin, please rerender

**Comment by chrisburr** [comment]
Re-rendering doesn't do anything on staged-recipes and I think you can ignore the Azure error as only AppVeyor for building the package (until that error is globally fixed at least).

**Comment by constantinpape** [comment]
> Re-rendering doesn't do anything on staged-recipes

Oh right, forgot about this, thanks.



> I think you can ignore the Azure error

Ok.

**Comment by constantinpape** [comment]
Just rebased on `conda-forge/master` to update the conda-forge configs.

**Comment by constantinpape** [comment]
Still having some build issues:
1. osx: The build uses c++14 (although the clang version should support c++17) and thus fails beacuse it falls back to boost::filesystem, which is not linked anymore.
2. win and linux: There is a numpy version conflict for python 3.7. On linux it looks like this, win is similar:
```
Package numpy-base conflicts for:
numpy=1.9 -> numpy-base[version='1.9.3|>=1.9.3,<2.0a0',build='py37h2b20989_7|py37h2b20989_6|py36hdbf6ddf_6|py36h2b20989_6|py35h2b20989_7|py27hdbf6ddf_6|py27h2b20989_6|py27h2b20989_7|py27hdbf6ddf_7|py35hdbf6ddf_7|py36h2b20989_7|py36hdbf6ddf_7|py37hdbf6ddf_6|py37hdbf6ddf_7']
xtensor-python[version='>=0.22,<0.23'] -> numpy -> numpy-base[version='1.11.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.4|1.14.4|1.14.4|1.14.4|1.14.4|1.14.4|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.   5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.  1|1.15.1|1.15.1|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.3|1.15.3|1.15.3|1.15.3|1.15.3|1.15.3|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.  0|1.16.0|1.16.0|1.16.0|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.   3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|>=1.9.3,<2.0a0',                                                                                                                                                                                                                                    build='py37h2b20989_7|py37h2b20989_6|py36hdbf6ddf_7|py36h2b20989_6|py35h2b20989_7|py27hdbf6ddf_7|py27hdbf6ddf_6|py37hde5b4d6_0|py36hde5b4d6_0|py27hde5b4d6_0|py36hde5b4d6_0|py37h2f8d375_0|py36hde5b4d6_0|py36h2f8d375_0|py27hde5b4d6_0|py27h2f8d375_0|py37hde5b4d6_1|py37h2f8d375_0|py36h2f8d375_1|py27h2f8d375_0|py37hde5b4d6_1|py37hde5b4d6_0|py37h2f8d375_0|py36h2f8d375_1|py27hde5b4d6_1|py27h2f8d375_1|py37hde5b4d6_0|py37h81de0dd_0|py37h2f8d375_0|py36hde5b4d6_0|py36h81de0dd_0|py27h81de0dd_0|py27h2f8d375_0|py37h81de0dd_0|py37h2f8d375_0|py36h81de0dd_0|py27h2f8d375_0|py37h81de0dd_1|py37h81de0dd_0|py36h81de0dd_1|py36h81de0dd_0|py35h2f8d375_0|py27h2f8d375_1|py37h81de0dd_0|py37h74e8950_0|py36h81de0dd_0|py35h81de0dd_0|py35h74e8950_0|py35h2f8d375_0|py37h7cdd4dd_0|py36h7cdd4dd_0|py36h3dfced4_0|py35h7cdd4dd_0|py37h81de0dd_4|py36h81de0dd_4|py35h81de0dd_4|py35h2f8d375_4|py27h81de0dd_4|py27h2f8d375_4|py37hdbf6ddf_3|py37hdbf6ddf_2|py37hdbf6ddf_1|py37h2b20989_4|py37h2b20989_3|py37h2b20989_1|py36hdbf6ddf_4|py36hdbf6ddf_0|py36h2b20989_4|py35hdbf6ddf_4|py35hdbf6ddf_0|py27hdbf6ddf_4|py27h2b20989_1|py27h2b20989_0|py36hdbf6ddf_0|py36h2b20989_0|py35hdbf6ddf_0|py35h2b20989_0|py27h2b20989_0|py36h0ea5e3f_1|py37hde5b4d6_11|py37hdbf6ddf_8|py37h81de0dd_10|py37h7cdd4dd_9|py37h74e8950_9|py37h74e8950_10|py37h3dfced4_9|py37h2f8d375_11|py37h2f8d375_10|py37h2b20989_7|py36hde5b4d6_12|py36hde5b4d6_11|py36hdbf6ddf_7|py36h81de0dd_9|py36h81de0dd_10|py36h7cdd4dd_9|py36h74e8950_9|py36h74e8950_10|py36h3dfced4_9|py36h2f8d375_11|py36h2f8d375_10|py36h2b20989_8|py35hdbf6ddf_8|py35h81de0dd_9|py35h81de0dd_10|py35h7cdd4dd_9|py35h74e8950_9|py35h3dfced4_9|py27hde5b4d6_11|py27hdbf6ddf_7|py27h81de0dd_9|py27h81de0dd_10|py27h74e8950_10|py27h2f8d375_12|py27h2b20989_8|py27h2b20989_7|py27h2f8d375_10|py27h2f8d375_11|py27h3dfced4_9|py27h74e8950_9|py27h7cdd4dd_9|py27hdbf6ddf_8|py27hde5b4d6_12|py35h2b20989_8|py35h2f8d375_10|py35h74e8950_10|py36h2b20989_7|py36h2f8d375_12|py36hdbf6ddf_8|py37h2b20989_8|py37h2f8d375_12|py37h81de0dd_9|py37hdbf6ddf_7|py37hde5b4d6_12|py27h0ea5e3f_1|py27h9be14a7_1|py35h0ea5e3f_1|py35h9be14a7_1|py36h9be14a7_1|py27hdbf6ddf_0|py27h2b20989_2|py27h2b20989_3|py27h2b20989_4|py27hdbf6ddf_0|py27hdbf6ddf_1|py27hdbf6ddf_2|py27hdbf6ddf_3|py35h2b20989_4|py36h2b20989_0|py36h2b20989_1|py36h2b20989_2|py36h2b20989_3|py36hdbf6ddf_1|py36hdbf6ddf_2|py36hdbf6ddf_3|py37h2b20989_2|py37hdbf6ddf_4|py36h2f8d375_4|py37h2f8d375_4|py27h3dfced4_0|py27h7cdd4dd_0|py35h3dfced4_0|py37h3dfced4_0|py27h2f8d375_0|py27h74e8950_0|py27h81de0dd_0|py36h2f8d375_0|py36h74e8950_0|py37h2f8d375_0|py27h2f8d375_0|py27h81de0dd_0|py27h81de0dd_1|py35h81de0dd_0|py36h2f8d375_0|py36h2f8d375_1|py37h2f8d375_0|py37h2f8d375_1|py27h81de0dd_0|py36h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py27h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py36hde5b4d6_0|py36hde5b4d6_1|py37h2f8d375_1|py27h2f8d375_1|py27hde5b4d6_0|py27hde5b4d6_1|py36h2f8d375_0|py36hde5b4d6_0|py36hde5b4d6_1|py37h2f8d375_1|py37hde5b4d6_0|py37hde5b4d6_0|py27h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py37h2f8d375_0|py37hde5b4d6_0|py27h2f8d375_0|py36h2f8d375_0|py37h2f8d375_0|py27h2b20989_6|py27h2b20989_7|py35hdbf6ddf_7|py36h2b20989_7|py36hdbf6ddf_6|py37hdbf6ddf_6|py37hdbf6ddf_7']
xtensor-python[version='>=0.22,<0.23'] -> numpy -> mkl_random[version='>=1.0.2,<2.0a0'] -> numpy-base[version='>=1.0.2,<2.0a0|>=1.0.6,<2.0a0']
z5py[version='>=1.5.1'] -> numpy[version='>=1.14.6,<2.0a0'] -> numpy-base[version='1.14.6|1.15.0|1.15.0|1.15.0|1.15.0|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.3|1.15.3|1.15.3|1.15.3|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1. 15.4|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.2|1.16.2|1.16.2|1.16.2|1.16.3|1.16.3|1.16.3|1.16.3|1.16.4|1.16.4|1.16.4|1.16.4',                                                                                build='py36h2f8d375_0|py37hde5b4d6_0|py36hde5b4d6_0|py36h2f8d375_0|py37hde5b4d6_1|py37h2f8d375_0|py36hde5b4d6_1|py36hde5b4d6_0|py36h2f8d375_0|py37hde5b4d6_1|py37hde5b4d6_0|py36h2f8d375_1|py36h2f8d375_0|py37h81de0dd_0|py36hde5b4d6_0|py36h81de0dd_0|py36h2f8d375_0|py37h81de0dd_0|py36h2f8d375_0|py37h81de0dd_1|py37h81de0dd_0|py36h81de0dd_1|py36h81de0dd_0|py36h2f8d375_1|py36h2f8d375_0|py37h81de0dd_0|py37h74e8950_0|py37h2f8d375_0|py36h81de0dd_0|py36h2f8d375_0|py37h3dfced4_0|py36h7cdd4dd_0|py36h3dfced4_0|py37h81de0dd_4|py36h81de0dd_4|py36h2f8d375_4|py37h2f8d375_4|py37h7cdd4dd_0|py36h74e8950_0|py37h2f8d375_0|py37h2f8d375_1|py36h81de0dd_0|py37h2f8d375_0|py37h2f8d375_0|py37hde5b4d6_0|py36hde5b4d6_0|py36hde5b4d6_1|py37h2f8d375_0|py37h2f8d375_1|py36h2f8d375_1|py37h2f8d375_1|py37hde5b4d6_0|py37h2f8d375_0|py37hde5b4d6_0|py36h2f8d375_0|py36hde5b4d6_0|py37h2f8d375_0|py36hde5b4d6_0|py37h2f8d375_0|py37hde5b4d6_0']
z5py[version='>=1.5.1'] -> numpy[version='>=1.14.6,<2.0a0'] -> mkl_fft -> numpy-base[version='>=1.0.2,<2.0a0|>=1.0.6,<2.0a0']
vigra -> numpy[version='>=1.9.3,<2.0a0'] -> numpy-base[version='1.11.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.3|1.14.4|1.14.4|1.14.4|1.14.4|1.14.4|1.14.4|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.   5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.5|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.14.6|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.0|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.1|1.15.  1|1.15.1|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.2|1.15.3|1.15.3|1.15.3|1.15.3|1.15.3|1.15.3|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.15.4|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.0|1.16.  0|1.16.0|1.16.0|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.1|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.2|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.3|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.16.4|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|1.9.3|>=1.9.3,<2.0a0',                                                                                                                                                                                                                                              build='py37h2b20989_7|py37h2b20989_6|py36hdbf6ddf_7|py36h2b20989_6|py35h2b20989_7|py27hdbf6ddf_7|py27hdbf6ddf_6|py37hde5b4d6_0|py36hde5b4d6_0|py27hde5b4d6_0|py36hde5b4d6_0|py37h2f8d375_0|py36hde5b4d6_0|py36h2f8d375_0|py27hde5b4d6_0|py27h2f8d375_0|py37hde5b4d6_1|py37h2f8d375_0|py36h2f8d375_1|py27h2f8d375_0|py37hde5b4d6_1|py37hde5b4d6_0|py37h2f8d375_0|py36h2f8d375_1|py27hde5b4d6_1|py27h2f8d375_1|py37hde5b4d6_0|py37h81de0dd_0|py37h2f8d375_0|py36hde5b4d6_0|py36h81de0dd_0|py27h81de0dd_0|py27h2f8d375_0|py37h81de0dd_0|py37h2f8d375_0|py36h81de0dd_0|py27h2f8d375_0|py37h81de0dd_1|py37h81de0dd_0|py36h81de0dd_1|py36h81de0dd_0|py35h2f8d375_0|py27h2f8d375_1|py37h81de0dd_0|py37h74e8950_0|py36h81de0dd_0|py35h81de0dd_0|py35h74e8950_0|py35h2f8d375_0|py37h7cdd4dd_0|py36h7cdd4dd_0|py36h3dfced4_0|py35h7cdd4dd_0|py37h81de0dd_4|py36h81de0dd_4|py35h81de0dd_4|py35h2f8d375_4|py27h81de0dd_4|py27h2f8d375_4|py37hdbf6ddf_3|py37hdbf6ddf_2|py37hdbf6ddf_1|py37h2b20989_4|py37h2b20989_3|py37h2b20989_1|py36hdbf6ddf_4|py36hdbf6ddf_0|py36h2b20989_4|py35hdbf6ddf_4|py35hdbf6ddf_0|py27hdbf6ddf_4|py27h2b20989_1|py27h2b20989_0|py36hdbf6ddf_0|py36h2b20989_0|py35hdbf6ddf_0|py35h2b20989_0|py27h2b20989_0|py36h0ea5e3f_1|py37hde5b4d6_11|py37hdbf6ddf_8|py37h81de0dd_10|py37h7cdd4dd_9|py37h74e8950_9|py37h74e8950_10|py37h3dfced4_9|py37h2f8d375_11|py37h2f8d375_10|py37h2b20989_7|py36hde5b4d6_12|py36hde5b4d6_11|py36hdbf6ddf_7|py36h81de0dd_9|py36h81de0dd_10|py36h7cdd4dd_9|py36h74e8950_9|py36h74e8950_10|py36h3dfced4_9|py36h2f8d375_11|py36h2f8d375_10|py36h2b20989_8|py35hdbf6ddf_8|py35h81de0dd_9|py35h81de0dd_10|py35h7cdd4dd_9|py35h74e8950_9|py35h3dfced4_9|py27hde5b4d6_11|py27hdbf6ddf_7|py27h81de0dd_9|py27h81de0dd_10|py27h74e8950_10|py27h2f8d375_12|py27h2b20989_8|py27h2b20989_7|py27h2f8d375_10|py27h2f8d375_11|py27h3dfced4_9|py27h74e8950_9|py27h7cdd4dd_9|py27hdbf6ddf_8|py27hde5b4d6_12|py35h2b20989_8|py35h2f8d375_10|py35h74e8950_10|py36h2b20989_7|py36h2f8d375_12|py36hdbf6ddf_8|py37h2b20989_8|py37h2f8d375_12|py37h81de0dd_9|py37hdbf6ddf_7|py37hde5b4d6_12|py27h0ea5e3f_1|py27h9be14a7_1|py35h0ea5e3f_1|py35h9be14a7_1|py36h9be14a7_1|py27hdbf6ddf_0|py27h2b20989_2|py27h2b20989_3|py27h2b20989_4|py27hdbf6ddf_0|py27hdbf6ddf_1|py27hdbf6ddf_2|py27hdbf6ddf_3|py35h2b20989_4|py36h2b20989_0|py36h2b20989_1|py36h2b20989_2|py36h2b20989_3|py36hdbf6ddf_1|py36hdbf6ddf_2|py36hdbf6ddf_3|py37h2b20989_2|py37hdbf6ddf_4|py36h2f8d375_4|py37h2f8d375_4|py27h3dfced4_0|py27h7cdd4dd_0|py35h3dfced4_0|py37h3dfced4_0|py27h2f8d375_0|py27h74e8950_0|py27h81de0dd_0|py36h2f8d375_0|py36h74e8950_0|py37h2f8d375_0|py27h2f8d375_0|py27h81de0dd_0|py27h81de0dd_1|py35h81de0dd_0|py36h2f8d375_0|py36h2f8d375_1|py37h2f8d375_0|py37h2f8d375_1|py27h81de0dd_0|py36h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py27h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py36hde5b4d6_0|py36hde5b4d6_1|py37h2f8d375_1|py27h2f8d375_1|py27hde5b4d6_0|py27hde5b4d6_1|py36h2f8d375_0|py36hde5b4d6_0|py36hde5b4d6_1|py37h2f8d375_1|py37hde5b4d6_0|py37hde5b4d6_0|py27h2f8d375_0|py27hde5b4d6_0|py36h2f8d375_0|py37h2f8d375_0|py37hde5b4d6_0|py27h2f8d375_0|py36h2f8d375_0|py37h2f8d375_0|py27h2b20989_6|py27h2b20989_7|py35hdbf6ddf_7|py36h2b20989_7|py36hdbf6ddf_6|py37hdbf6ddf_6|py37hdbf6ddf_7']
Package numpy conflicts for:
vigra -> numpy[version='>=1.9.3,<2.0a0'] -> mkl_random[version='>=1.0.2,<2.0a0'] -> numpy[version='>=1.11.3,<2.0a0']
vigra -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|>=1.8|>=1.9.3,<2.0a0']
xtensor-python[version='>=0.22,<0.23'] -> numpy
numpy=1.9
z5py[version='>=1.5.1'] -> numpy[version='>=1.14.6,<2.0a0']
```

**Comment by constantinpape** [comment]
Re 2.:
I cannot reproduce this locally: I can create the following environment without version conflicts:
```
name:
  nifty-dep-test
channels:
  - conda-forge
dependencies:
  - numpy
  - xtensor>=0.19,<0.20
  - xtensor-python>=0.22,<0.23
  - vigra
  - hdf5>=1.10
  - z5py>=1.5.1
  - nlohmann_json
  - blosc
  - bzip2
  - zlib
  - scikit-image
```
with the following setup:
```
$ conda info
     active environment : None
            shell level : 0
       user config file : /home/pape/.condarc
 populated config files : 
          conda version : 4.6.14
    conda-build version : 3.17.8
         python version : 3.7.1.final.0
       base environment : /home/pape/Work/software/conda/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/pape/Work/software/conda/miniconda3/pkgs
                          /home/pape/.conda/pkgs
       envs directories : /home/pape/Work/software/conda/miniconda3/envs
                          /home/pape/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.14 requests/2.21.0 CPython/3.7.1 Linux/5.0.0-23-generic ubuntu/18.04.2 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

I rebased on `conda-forge/master` very recently in order to bring the build configuration up-to-date, but I start to suspect that this did not quite work out. Is there any way of checking this?
Any other ideas what might cause these issues?

**Pull request closed without merging**